damus/Views/Video/VideoPlayer.swift

```diff
@@ -166,27 +188,33 @@ public extension VideoPlayer {
     }
 }
 
-func get_video_size(player: AVPlayer) async -> CGSize? {
-    let res = Task.detached(priority: .background) {
-        return player.currentImage?.size
+fileprivate extension AVPlayer {
+    var videoSize: CGSize? {
+        get async {
+            let res = Task.detached(priority: .background) {
```

Comment:
It's not a good idea to use `.background` priority for something that you want to update the UI for the user. I would recommend using the default value, which is `.userInitiated`.